File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/566. Reshape the Matrix/C++ solution/566. Reshape the Matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        int R = mat.size(), C = mat[0].size();
        int size = R * C;

        if (size != r * c || (R == r && C == c))
            return mat;

        vector<vector<int>> reshaped(r, vector<int>(c));
        vector <int> mat_liner(size);

        for (int i = 0; i < R; i++)
        {
            for (int j = 0; j < C; j++)
            {
                mat_liner[(C * i) + j] = mat[i][j];
            }
        }

        for (int i = 0; i < size; i++)
        {
            reshaped[i / c][i % c] = mat_liner[i];
        }
        
        return reshaped;
    }
};

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/566. Reshape the Matrix/Python solution/566. Reshape the Matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        reshaped = [[0] * c for _ in range(r)] # initialize a 2D array
        size = len(mat) * len(mat[0])
        mat_liner = [0] * size

        if size != r * c or (len(mat) == r and len(mat[0]) == c):
            return mat

        for i in range(len(mat)):
            for j in range(len(mat[0])):
                mat_liner[(len(mat[0]) * i) + j] = mat[i][j]

        for s in range(size):
            reshaped[s//c][s%c] = mat_liner[s]
            
        return reshaped